packages/fcl-wc/src/service.js

```diff
@@ -0,0 +1,170 @@
+import QRCodeModal from "@walletconnect/qrcode-modal"
+
+export const makeServicePlugin = client => ({
+  name: "fcl-service-walletconnect",
+  type: "discovery-service",
+  services: makeWcServices(client),
+})
+
+const makeServiceStrategy = client => {
+  return ({service, body, opts}) => {
+    return new Promise(async (resolve, reject) => {
+      if (typeof client === "undefined") {
+        throw new Error("WalletConnect is not initialized")
+      }
+      let session
+      const onResponse = resp => {
+        try {
+          if (typeof resp !== "object") return
+
+          switch (resp.status) {
+            case "APPROVED":
+              resolve(resp.data)
+              break
+
+            case "DECLINED":
+              reject(`Declined: ${resp.reason || "No reason supplied"}`)
+              break
+
+            case "REDIRECT":
+              resolve(resp)
+              break
+
+            default:
+              reject(`Declined: No reason supplied`)
+              break
+          }
+        } catch (error) {
+          console.error("execWcRPC onResponse error", error)
+          throw error
+        }
+      }
+
+      const onClose = () => {
+        reject(`Declined: Externally Halted`)
+      }
+
+      if (client.session.length) {
```

Comment:
Session looks like an object and you're calling length on it...think you need `Object.keys`

Reply:
So this is weird as well. 
The `session` stored in localStorage is an array of `topic` strings.
[Their example code](https://github.com/WalletConnect/web-examples/blob/main/dapps/react-dapp-v2/src/contexts/ClientContext.tsx#L218) uses client.session.length to check for existing session and use the last session (`const lastKeyIndex = client.session.keys.length - 1`) 
`client.session.length` and `client.session.keys` return number of topics saved in client localStorage correctly.
Object.keys(client.session).length returns the length of keys in the object (15)